### ISSUE TITLE
Large banners need mobile versions

### ISSUE THREAD

**Issue by joshistoast** [Shopify/dawn#3195]
## Describe the enhancement you'd like

One feature that's a must for Shopify themes (especially in my company's usage) is for banners, slideshows, and slideshows there should be an option for mobile imagery.
This would be good for showing landscape images on desktop, and optimized square/portrait imagery for mobile viewing. It's annoying having a wide landscape image on desktop be handled so poorly on mobile.
Right now dawn offers 2 images and this is in my opinion very poorly implemented and provides little utility.

## Describe alternatives you've considered
I could add this behavior myself but I'd rather this be a native feature.

## Additional context/screenshots
<img width="465" alt="Screenshot 2024-01-16 at 3 17 30 PM" src="https://github.com/Shopify/dawn/assets/23065423/1423eeb7-90e1-4370-a515-a2867830b460">

**Comment by SmolSoftBoi** [comment]
Could you add a pull request with this behaviour?

**Comment by joshistoast** [comment]
I agree in failing to see any valid use case for 2 images in the way dawn expects. Thanks for the code snippet @gregjotau! I'll have to try it, I was building out a separate hero section due to the existing styling and hierarchical complexity of the existing image banner component, starting fresh would be easier haha.

**Comment by aaannorst** [comment]
Please, somebody enlighten me, 
as I read **image-banner.liquid** it seems that the banner's default image is 1500px wide, all other (/smaller) sizes come later. Is that correct? If is, this section doesn't reflect the _mobile first_ design approach. 
I would rather use a default config file with breakpoints link-able to any sections / enhanced by use cases. 

```
      {{ section.settings.image | image_url: width: 1500 | image_tag:
        loading: 'eager',
        width: section.settings.image.width,
        height: image_height,
        class: image_class,
        sizes: sizes,
        widths: '375, 550, 750, 1100, 1500, 1780, 2000, 3000, 3840',
        alt: section.settings.image.alt | escape
      }}
```

**Comment by joshistoast** [comment]
I addressed this by adding a "Visibility" setting to relevant sections, with the options: `Desktop`, `Mobile` or `Both`. I see this being a neat addition to Dawn.
